### PR TITLE
docs(root): change in contribution docs

### PR DESCRIPTION
**Pull request by gd2910**
## Summary of the changes
Changes to contribution guidance regarding testing with cypress.

## Related issue
#1270 

## Checklist
- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] All acceptance criteria reviewed and met. 
- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.
- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] Browser support tested (Chrome, Safari, Firefox and Edge).
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 
- [x] All prop combinations work without issue. 
- [x] Changes to docs package checked and committed.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
 ### End-to-end / integration tests
 
-Stencil provides utility functions from [Jest](https://jestjs.io/) and [Puppeteer](https://pptr.dev/) to create end-to-end / integration tests. End-to-end / integraton testing makes it easier to test multiple components within a real browser. Each component includes a `<component>.e2e.ts` file.
+- e2e tests and how to use selectors etc
+- where to put the tests
+- a11y tests
+- visual tests
+
+
+[Cypress](https://www.cypress.io/) is the testing framework used to run end-to-end / intergration tests. Within the `react/src/component-tests` directory is a folder for each component that includes a `<component>.cy.tsx` file along with any helper data for the test.
```
Comment (GCHQ-Developer-299) [inline CONTRIBUTING.md:360]: I would like to see a bit more depth on the file structure - we've got a pattern with tests in component.cy.tsx, componentConstants.tsx to store values for testing against, and componentTestData.tsx for storing code snippets to improve test readability.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+    cy.get("ic-typogrpahy").should("have.text", "baz")
   });
 ```
 
 ### Visual regression tests
 
-[Loki](https://loki.js.org/) runs visual regression tests by building Storybook and taking snapshots of each story. As updates are made to components, the reference images are compared to the latest Storybook build to generate a pixel-to-pixel comparison and flag any differences. All reference images are stored within `.loki` directory. Loki uses Google Chrome to render the components.
+[Cypress](https://www.cypress.io/) itself does not have built-in support for visual regression testing, however the third-party plugin `cypress-image-diff-js` adds this functionality. The tool works by taking screenshots of the components at specific points during tests and then comparing them to the baseline images to detect any visual differences. All reference images are stored within the `cypress-visual-screenshots` directory once the test suite is ran. Cypress uses a headless Electron browser for running tests and rendering components.
+
+### A11y tests
```
[the remark this answers is not in the document]
Comment (GCHQ-Developer-299) [reply CONTRIBUTING.md:391]: One of our colleagues has suggested we include some best practice information, this heading could be removed and then A11y tests described in that section maybe?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+    // Check the component has hydrated to avoid basic errors
+    cy.checkHydrated("ic-button");
+
+    cy.compareSnapshot("default", DEFAULT_TEST_THRESHOLD);
+  });
+```
+
+### Testing best practice
+
+- ensure all props are tested.
```
Comment (GCHQ-Developer-465) [inline CONTRIBUTING.md:402]:
```suggestion
- Ensure all props are tested.

```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+- visual tests should include sufficient padding to allow for focus indicators (and there should be screenshots for showing items when focused).
```
Comment (GCHQ-Developer-465) [inline CONTRIBUTING.md:403]:
```suggestion
- Visual tests should include sufficient padding to allow for focus indicators (and there should be screenshots for showing items when focused).

```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+- snapshots should be added for any props that affect the visual display of the component.
```
Comment (GCHQ-Developer-465) [inline CONTRIBUTING.md:404]:
```suggestion
- Snapshots should be added for any props that affect the visual display of the component.

```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+- a11y tests should be included.
```
Comment (GCHQ-Developer-465) [inline CONTRIBUTING.md:405]:
```suggestion
- A11y tests should be included.

```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -351,36 +351,59 @@ Read more about [Stencil Unit testing](https://stenciljs.com/docs/unit-testing).
...
+- ensure all tests are passing consistently, both locally and in CI.
```
Comment (GCHQ-Developer-465) [inline CONTRIBUTING.md:406]:
```suggestion
- Ensure all tests are passing consistently, both locally and in CI.

```

**Review by GCHQ-Developer-465 — CHANGES_REQUESTED** [review]
Just some grammar :) Looks great though!

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by GCHQ-Developer-465 — APPROVED** [review]

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request merged**